auth_requestor_systemd_test.go

```diff
@@ -55,219 +57,180 @@ func (s *authRequestorSystemdSuite) setPassphrase(c *C, passphrase string) {
 
 var _ = Suite(&authRequestorSystemdSuite{})
 
-type testRequestPassphraseData struct {
+type testRequestUserCredentialParams struct {
 	passphrase string
 
-	tmpl string
-
-	volumeName       string
-	sourceDevicePath string
+	ctx       context.Context
+	name      string
+	path      string
+	authTypes UserAuthType
 
 	expectedMsg string
 }
 
-func (s *authRequestorSystemdSuite) testRequestPassphrase(c *C, data *testRequestPassphraseData) {
-	s.setPassphrase(c, data.passphrase)
+func (s *authRequestorSystemdSuite) testRequestUserCredential(c *C, params *testRequestUserCredentialParams) {
+	s.setPassphrase(c, params.passphrase)
 
-	requestor := NewSystemdAuthRequestor(data.tmpl, "")
+	requestor := NewSystemdAuthRequestor(map[UserAuthType]string{
+		UserAuthTypePassphrase:                                             "Enter passphrase for %[1]s (%[2]s):",
+		UserAuthTypePIN:                                                    "Enter PIN for %[1]s (%[2]s):",
+		UserAuthTypeRecoveryKey:                                            "Enter recovery key for %[1]s (%[2]s):",
+		UserAuthTypePassphrase | UserAuthTypePIN:                           "Enter passphrase or PIN for %[1]s (%[2]s):",
+		UserAuthTypePassphrase | UserAuthTypeRecoveryKey:                   "Enter passphrase or recovery key for %[1]s (%[2]s):",
+		UserAuthTypePIN | UserAuthTypeRecoveryKey:                          "Enter PIN or recovery key for %[1]s (%[2]s):",
+		UserAuthTypePassphrase | UserAuthTypePIN | UserAuthTypeRecoveryKey: "Enter passphrase, PIN or recovery key for %[1]s (%[2]s):",
```

Comment:
So we have to provide 2^N-1 strings. If we two different kind of keys, that will become 31 strings? I do not know if there are any clever way to handle it, but I feel it would be nice if we could capture at compile time if we are missing some cases.